### ISSUE TITLE
unresolved dependency, configuration not found 'default'

### ISSUE THREAD

**Issue by easel** [SparklineData/spark-datetime#7]
This may just be operator error on my part, but I'm not able to get spark-datetime to work with a downloaded spark 1.4.1 scala 2.10 or locally compiled with scala 2.11. Consistently, I get the following error, which _appears_ to be related to some downstream dependency that isn't clearly called out in the error message.

I'm starting it via spark shell: `bin/spark-shell --packages org.sparklinedata:spark-datetime_2.10:0.0.1`

```
Exception in thread "main" java.lang.RuntimeException: [unresolved dependency: org.sparklinedata#spark-datetime_2.10;0.0.1: configuration not found in org.sparklinedata#spark-datetime_2.10;0.0.1: 'default'. It was required from org.apache.spark#spark-submit-parent;1.0 default]
    at org.apache.spark.deploy.SparkSubmitUtils$.resolveMavenCoordinates(SparkSubmit.scala:995)
    at org.apache.spark.deploy.SparkSubmit$.prepareSubmitEnvironment(SparkSubmit.scala:263)
    at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:145)
    at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:112)
    at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
```

**Comment by hbutani** [comment]
This is a known issue. 
I have not been successful in registering a release on the spark-packages site.  Trying to find someone from databricks to help with this.
In the meantime please do a local build and use '--jars' option of spark-shell.

**Comment by easel** [comment]
Ok, I started down that path but gave up after the 2nd or third jar. I'm sure there's an end to the rabbit hole somewhere =p Thanks for the reply!

**Comment by hbutani** [comment]
You shouldn't have to pull in any other dependencies, if you use the assembly jar. 
If you run sbt assembly it creates this; then just use it in the jars param of spark-shell.

If this is not working for you, let me know; I will look into it right away.

**Comment by easel** [comment]
Ahh, the assembly jar. I was able to get it to work with this: bin/spark-shell --packages com.databricks:spark-csv_2.10:1.1.0 --jars $HOME/.ivy2/local/org.sparklinedata/spark-datetime_2.10/0.0.1/jars/spark-datetime_2.10.jar,$HOME/.ivy2/cache/joda-time/joda-time/jars/joda-time-2.8.1.jar,$HOME/.ivy2/cache/com.github.nscala-time/nscala-time_2.10/jars/nscala-time_2.10-1.6.0.jar

Attempting to compile the assembly jar results in a bunch of test errors: 

```
[spark-datetime] sbt clean +assembly                                                                                                                                                                             master
[info] Loading global plugins from /Users/erik/.sbt/0.13/plugins
[info] Loading project definition from /Users/erik/Projects/spark-datetime/project
[info] Set current project to spark-datetime (in build file:/Users/erik/Projects/spark-datetime/)
[warn] Credentials file /Users/erik/.github.cred does not exist
[success] Total time: 0 s, completed Aug 1, 2015 10:07:38 AM
[info] Setting version to 2.10.4
[info] Reapplying settings...
[info] Set current project to spark-datetime (in build file:/Users/erik/Projects/spark-datetime/)
[warn] Credentials file /Users/erik/.github.cred does not exist
[info] Updating {file:/Users/erik/Projects/spark-datetime/}spark-datetime...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Compiling 4 Scala sources to /Users/erik/Projects/spark-datetime/target/scala-2.10/classes...
[info] Compiling 3 Scala sources to /Users/erik/Projects/spark-datetime/target/scala-2.10/test-classes...
[info] Including: nscala-time_2.10-1.6.0.jar
[info] Including: joda-convert-1.2.jar
[info] Including: joda-time-2.5.jar
[error] bad symbolic reference. A signature in package.class refers to type compileTimeOnly
[error] in package scala.annotation which is not available.
[error] It may be completely missing from the current classpath, or the version on
[error] the classpath might be incompatible with the version used when compiling package.class.
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/BaseTest.scala:28: Reference to class FunSuite in package scalatest should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error] abstract class BaseTest extends FunSuite with BeforeAndAfterAll {
[error]                                 ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/BaseTest.scala:41: Reference to method createDataFrame in class SQLContext should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]     val df = createDataFrame[TRow](sparkContext.parallelize(col))
[error]              ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/BaseTest.scala:41: Reference to method getClassLoader in class Class should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]     val df = createDataFrame[TRow](sparkContext.parallelize(col))
[error]                                   ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/BaseTest.scala:48: Reference to method createDataFrame in class SQLContext should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]     val df1 = createDataFrame[TRow](sparkContext.parallelize(col1))
[error]               ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/BaseTest.scala:48: Reference to method getClassLoader in class Class should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]     val df1 = createDataFrame[TRow](sparkContext.parallelize(col1))
[error]                                    ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:170: Reference to method getID in class DateTimeZone should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       assert(oDt.getZone.getID == timeZoneIdVal)
[error]                          ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:210: Reference to method -> in class ArrowAssoc should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Monday" -> 5,
[error]                ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:210: Reference to method any2ArrowAssoc in object Predef should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Monday" -> 5,
[error]       ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:211: Reference to method -> in class ArrowAssoc should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Tuesday" -> 5,
[error]                 ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:211: Reference to method any2ArrowAssoc in object Predef should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Tuesday" -> 5,
[error]       ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:212: Reference to method -> in class ArrowAssoc should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Wednesday" -> 4,
[error]                   ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:212: Reference to method any2ArrowAssoc in object Predef should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Wednesday" -> 4,
[error]       ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:213: Reference to method -> in class ArrowAssoc should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Friday" -> 4,
[error]                ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:213: Reference to method any2ArrowAssoc in object Predef should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Friday" -> 4,
[error]       ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:214: Reference to method -> in class ArrowAssoc should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Sunday" -> 4,
[error]                ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:214: Reference to method any2ArrowAssoc in object Predef should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Sunday" -> 4,
[error]       ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:215: Reference to method -> in class ArrowAssoc should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Thursday" -> 4,
[error]                  ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:215: Reference to method any2ArrowAssoc in object Predef should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Thursday" -> 4,
[error]       ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:216: Reference to method -> in class ArrowAssoc should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Saturday" -> 4)
[error]                  ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:216: Reference to method any2ArrowAssoc in object Predef should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]       "Saturday" -> 4)
[error]       ^
[error] /Users/erik/Projects/spark-datetime/src/test/scala/org/sparklinedata/spark/dateTime/FunctionsTest.scala:248: Reference to method toString in class AbstractDateTime should not have survived past type checking,
[error] it should have been processed and eliminated during expansion of an enclosing macro.
[error]     val start = dateTime(START_DATE.toString)
[error]                                     ^
[error] 22 errors found
[error] (test:compile) Compilation failed
[error] Total time: 11 s, completed Aug 1, 2015 10:07:49 AM
[spark-datetime]
```

**Comment by hbutani** [comment]
Hmm
Which version of sbt?
sbt 0.13.6 works for me.

But I noticed build/sbt doesn't work. I get an 'Invalid or corrupt jarfile' error. Get the same behavior in spark-csv, have filed an issue https://github.com/databricks/spark-csv/issues/114

**Comment by hbutani** [comment]
Fixed in commit: bbd2d1fe21c 
Based on suggestions in databricks/spark-csv#114
